driver/k8s/logicalvolume_service.go

```diff
@@ -140,7 +140,28 @@ func (s *LogicalVolumeService) DeleteVolume(ctx context.Context, volumeID string
 		return err
 	}
 
-	return s.Delete(ctx, lv)
+	err = s.Delete(ctx, lv)
+	if err != nil {
+		return err
+	}
+
+	// wait until delete the target volume
+	for {
+		logger.Info("waiting for delete LogicalVolume", "name", lv.Name)
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-time.After(100 * time.Millisecond):
+		}
+
+		err := s.Get(ctx, client.ObjectKey{Name: lv.Name}, new(topolvmv1.LogicalVolume))
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return nil
+			}
+			logger.Error(err, "failed to get LogicalVolume", "name", lv.Name)
```

Comment:
I think it is better to return an error immediately when getting the error without IsNotFound.

Reply:
@llamerada-jp Why?  For me, this seems legit.